src/toil/bus.py

```diff
@@ -145,6 +155,18 @@ class JobAnnotationMessage(NamedTuple):
     # The annotation data
     annotation_value: str
 
+class BatchSystemMessage(NamedTuple):
```

Comment:
I'm not sure about the name here. All the other messages are named after events they represent (like `JobFailedMessage` or after data they are there to carry (like `QueueSizeMessage`). This one is named after the batch system, but it doesn't exist to tell you what the batch system is (though that is in there). It exists to tell you what the external batch system/backing scheduler's ID is for a particular job. 

Maybe this should be an `ExternalBatchIdMessage`? Or `ExternalIdMessage`?